app/lib/dfe/wizard.rb

```diff
@@ -0,0 +1,138 @@
+module DfE
+  class Wizard
+    attr_reader :current_step_name, :steps, :request
+    attr_writer :step_params
+
+    delegate :next_step, to: :current_step
+    delegate :info, to: :logger, allow_nil: true
+
+    def initialize(current_step:, step_params: {}, **args)
+      @current_step_name = current_step
+      @step_params = step_params
+      @steps = self.class.steps
+
+      (args || {}).each do |key, value|
+        send("#{key}=", value) if respond_to?("#{key}=")
+      end
+    end
+
+    def self.steps
+      return @steps unless block_given?
+
+      @steps = yield
+    end
+
+    def self.store(store_service = nil)
+      @store ||= store_service
+    end
+
+    def logger; end
+
+    def current_step
+      instance_current_step if step_form_object_class.present?
+    end
+
+    def valid_step?
+      current_step.valid?
+    end
+
+    def invalid_step?
+      !valid_step?
+    end
+
+    def save
+      store.save if store.present?
+    end
+
+    def store
+      @store ||= self.class.store.new(self) if self.class.store.present?
+    end
+
+    def step_params
+      if @step_params.respond_to?(:permit) && @step_params.key?(current_step_name)
+        @step_params.require(current_step_name).permit(permitted_params)
+      else
+        @step_params[current_step_name]
+      end
+    end
+
+    def instance_current_step
+      info("Instantiate steps with: #{current_step_params}") if @instance_current_step.blank?
+
+      @instance_current_step ||= step_form_object_class.new(current_step_params.merge(wizard: self))
+    end
+
+    def step_form_object_class
```

Comment:
Why is the word "form" in the method name?

Reply:
Could catch, let me change to `step_object_class`